backend/typescript/migrations/2024.10.22T19.38.07.create-behaviour-level-details-table.ts

```diff
@@ -0,0 +1,42 @@
+import { DataType } from "sequelize-typescript";
+
+import { Migration } from "../umzug";
+
+const TABLE_NAME = "behaviour_level_details";
+
+export const up: Migration = async ({ context: sequelize }) => {
+  await sequelize.getQueryInterface().createTable(TABLE_NAME, {
+    id: {
+      type: DataType.INTEGER,
+      allowNull: false,
+      primaryKey: true,
+      autoIncrement: true,
+    },
+    behaviour_id: {
+      type: DataType.INTEGER,
+      allowNull: false,
+    },
+    level: {
+      type: DataType.INTEGER,
+      allowNull: false,
+    },
+    description: {
+      type: DataType.STRING,
+      allowNull: false,
+    },
+    training_instructions: {
+      type: DataType.STRING,
+      allowNull: false,
+    },
+  });
+
+  await sequelize.getQueryInterface().addConstraint(TABLE_NAME, {
+    fields: ["behaviour_id", "level"],
+    type: "unique",
+    name: "unique_behaviour_level",
+  });
+};
+
+export const down: Migration = async ({ context: sequelize }) => {
+  await sequelize.getQueryInterface().dropTable(TABLE_NAME);
```

Comment:
I think you might need to drop the constraint too? I know things can get messy with the ordering of which you drop the constraints and the tables it is associated with.

You can play around with this, but the general success criteria is that you want to be able to run the up script and then the down script for your migration file, and the database should stay exactly the same